### PR TITLE
Velero workload

### PR DESCRIPTION
**Pull request by pranavgaikwad**
##### SUMMARY
New workload for Velero deployment

##### ISSUE TYPE
None

##### COMPONENT NAME
Workload

##### ADDITIONAL INFORMATION
Deploys Velero, Restic and Velero Migration Plugins as a workload on an OCP cluster with S3 as a backup storage.

### CODE REVIEW

File: ansible/roles/ocp-workload-velero/templates/restic-daemonset.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          secret:
+            secretName: {{ velero_aws_secret_name }}
+        - name: host-pods
+          hostPath:
+            path: /var/lib/kubelet/pods
+        - name: scratch
+          emptyDir: {}
+      containers:
+        - name: velero
+          image: quay.io/ocpmigrate/velero:fusor-dev
```
Comment (jwmatthews) [inline ansible/roles/ocp-workload-velero/templates/restic-daemonset.yml.j2]: +1 on having this use the images we've patched.

I think it'd be worth making this a variable in defaults that can be overridden

File: ansible/roles/ocp-workload-velero/templates/velero-deployment.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "8085"
+        prometheus.io/path: "/metrics"
+    spec:
+      restartPolicy: Always
+      serviceAccountName: velero
+      containers:
+        - name: velero
+          image: quay.io/ocpmigrate/velero:fusor-dev
```
Comment (jwmatthews) [inline ansible/roles/ocp-workload-velero/templates/velero-deployment.yml.j2]: I think it'd be worth making this a variable in defaults that can be overridden

File: ansible/roles/ocp-workload-velero/tasks/post_workload.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+- name: "Installing ocpmigrate plugin"
+  shell: "{{ velero_download_dir }}/velero/velero plugin add quay.io/ocpmigrate/velero-plugin"
```
Comment (jwmatthews) [inline ansible/roles/ocp-workload-velero/tasks/post_workload.yml]: Wonder if we can skip downloading velero release and just install plugin as below example does?

https://github.com/fusor/mig-controller/blob/master/hack/deploy/manifests/20-velero-deployment.yaml#L65-L74

**Review by jwmatthews — COMMENTED** [review]
Pranav, this is looking great.

File: ansible/roles/ocp-workload-velero/templates/velero-deployment.yml.j2 (hunk 2 above)
Comment (pranavgaikwad) [reply ansible/roles/ocp-workload-velero/templates/velero-deployment.yml.j2]: yeah. that makes sense. Will do that.

**Pull request closed without merging**